scripts/benchmark.py

```diff
@@ -0,0 +1,475 @@
+            if args.old != '':
+                revold = args.old
+            if args.new != '':
+                revnew = args.new
+
+            # load benchmark objects for each revision+bench_name combo for
+            bench_from = db.list(revold)
+            bench_to = db.list(revnew)
+            old_times = collections.OrderedDict()
+            new_times = collections.OrderedDict()
+            for old, new in zip(bench_from, bench_to):
+                old_times[old[1]] = db.load(old[0], old[1])
+                new_times[new[1]] = db.load(new[0], new[1])
+
+            print(BenchComp.header(revold, revnew))
+            for name in old_times:
+                if name not in new_times:
+                    continue
+
+                # print before-and-after comparison for each benchmark
+                old = old_times[name]
+                new = new_times[name]
+                if len(old.realruns) > 0 and len(new.realruns) > 0:
+                    cmp = BenchComp(old, new, usertime=args.usertime)
+                    print(cmp)
+            print(BenchComp.footer())
+
+def buildpage(fname, plotnames):
+    figpage = """
+<!DOCTYPE html>
+<meta charset="utf-8">
+<html>
+<head>
+    <style>
+        .flex-grid {
+            /*margin: 0 auto;*/
+            /*float: left;*/
+            /*width: 25%;*/
+            display: flex;
+        }
+        .col {
+            flex: 1;
+        }
+        .center {
+            text-align: center;
+        }
+        img.center {
+            display: block;
+            margin-left: auto;
+            margin-right: auto;
+        }
+    </style>
+</head>
+<body>
+    <div class="flex-grid">
+        {% for name in benchnames %}
+        <div class="col">
+            <img class="center" src="{{name}}.svg">
+            <h3 class="center">{{name}}</h3>
+        </div>
+        {% endfor %}
+    </div>
+</body>
+</html>
+"""
+
+    t = jinja2.Template(figpage)
+    data = t.render(benchnames=plotnames)
+    with open(fname, 'w') as f:
+        f.write(data)
+
+def plot(revisions, benchmarks, subdir='.'):
+    data = []
+    labels = []
+    for rev, bench in zip(revisions, benchmarks):
+        data.append(bench.realruns)
+        labels.append(rev[:6])
+
+    plt.boxplot(data, labels=labels)
+    plt.xticks(rotation=90)
+
+    fig = plt.gcf()
+    fig.subplots_adjust(bottom=.15)
+    fig.savefig(os.path.join(subdir, benchmarks[0].name + '.svg'))
+    plt.clf()
+    #plt.show()
+
+class Test:
+    def __init__(self, executable, infile, rootdir='.', args=None):
+        self.rootdir = rootdir
+        self.executable = executable
+        self.infile = infile
+        self.args = args
+        self.dur_secs = 0
+        self.user_dur_secs = 0
+        self.perflog = []
+
+    def run(self):
+        cmdpath = os.path.abspath(os.path.join(self.rootdir, self.executable))
+        infilepath = os.path.abspath(os.path.join(self.rootdir, self.infile))
+        cmd = [cmdpath, '-i', infilepath]
```

Comment:
This is rather limited. The Tester base class handles building the right command up and can deal with a variety of MPI/threading environment. I'm just pointing that out now because it's already done.